### ISSUE TITLE
Use basic numbers instead of UD60x18 for the percentage calculations

### ISSUE THREAD

**Issue by PaulRBerg** [sablier-labs/v2-periphery#317]
Using UD60x18 is overkill here - it's more gas expensive than `uint64` and doesn't offer any practical benefits:

https://github.com/sablier-labs/v2-periphery/blob/cb2ed635769dbe49d57fe8e297d8b77ac86450c5/src/SablierV2MerkleLockupFactory.sol#L71-L86

From the PRBMath [README](https://github.com/PaulRBerg/prb-math/blob/a111d11988756e091c7ccfd1f5af490d4ddf8783/README.md):

> it is important to be aware that utilizing these functions may result in increased gas costs compared to unwrapping and directly using the vanilla types

**Comment by smol-ninja** [comment]
I agree with this. Then we should also use `uint64` in `TrancheWithPercentage` struct. It would save gas too by avoiding [unwrapping](https://github.com/sablier-labs/v2-periphery/blob/0b22fed475134d4a9d638e7f19fade49d0a31bfe/src/SablierV2MerkleLockupFactory.sol#L75) from `UD2x18` to `uint64`.

https://github.com/sablier-labs/v2-periphery/blob/cb2ed635769dbe49d57fe8e297d8b77ac86450c5/src/types/DataTypes.sol#L120-L124

**Comment by PaulRBerg** [comment]
I'm fine to keep using `UD2x18` there because of the security benefits. Users should beware that that is a fixed-point number.
